llvm/lib/Target/RISCV/RISCVTargetMachine.cpp

```diff
@@ -599,6 +602,10 @@ void RISCVPassConfig::addMachineSSAOptimization() {
 }
 
 void RISCVPassConfig::addPreRegAlloc() {
+  // TODO: Move this as late as possible before regalloc
+  if (TM->getOptLevel() == CodeGenOptLevel::None)
```

Comment:
At -O0 we don't run `addMachineSSAOptimization` so we still need to eliminate them here